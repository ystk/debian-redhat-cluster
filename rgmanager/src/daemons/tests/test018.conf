<?xml version="1.0"?>
<!--
	while testing for #212121, I found that if you had multiple
	instances of untyped children where the untyped children were 
	multi-instance resources, you could end up with resource duplication
	the second time around.

        For example:

	start test2
	start initscript
	start clusterfs    - this should not happen twice
	start clusterfs
	start ip .1.3
	start mount2
	start dummy export
	start admin group
	start user group
	start red
	start script2
	start .1.4
	start script3

        ... would occur without the change to restree.c which removes
	the addition of newchild to the known-children.
	
-->
<cluster>
<rm>
	<resources>
    		<service name="test1"/>
    		<service name="test2"/>
		<script name="initscript" file="/etc/init.d/sshd"/>
		<script name="script2" file="/etc/init.d/script2"/>
		<script name="script3" file="/etc/init.d/script3"/>
		<ip address="192.168.1.3" monitor_link="yes"/>
		<ip address="192.168.1.4" monitor_link="yes"/>
    		<fs fstype="ext3" name="mount1" mountpoint="/mnt/cluster" device="/dev/sdb8"/>
    		<fs fstype="ext3" name="mount2" mountpoint="/mnt/cluster2" device="/dev/sdb9"/>
		<nfsexport name="Dummy Export"/>
    		<nfsclient name="User group" target="@users" options="rw,sync"/>
    		<nfsclient name="Admin group" target="@admin" options="rw"/>
    		<nfsclient name="yellow" target="yellow" options="rw,no_root_squash"/>
    		<nfsclient name="magenta" target="magenta" options="rw,no_root_squash"/>
    		<nfsclient name="red" target="red" options="rw"/>
		<clusterfs name="argle" mountpoint="/mnt/cluster3" device="/dev/sdb10"/>
		
	</resources>
	<service ref="test1">
		<script ref="initscript">
			<clusterfs ref="argle"/>
		</script>
		<fs ref="mount1">
			<nfsexport ref="Dummy Export">
				<nfsclient ref="Admin group"/>
				<nfsclient ref="User group"/>
				<nfsclient ref="red"/>
			</nfsexport>
		</fs>
	</service>
	<service ref="test2">
		<script ref="initscript">
			<clusterfs ref="argle"/>
			<ip ref="192.168.1.3"/>
			<fs ref="mount2">
				<nfsexport ref="Dummy Export">
					<nfsclient ref="Admin group"/>
					<nfsclient ref="User group"/>
					<nfsclient ref="red"/>
				</nfsexport>
			</fs>
			<script ref="script2"/>
			<ip ref="192.168.1.4"/>
		</script>
		<script ref="script3"/>
	</service>
</rm>
</cluster>
