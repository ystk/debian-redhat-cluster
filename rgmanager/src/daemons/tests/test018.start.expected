Starting test1...
[start] service:test1
[start] fs:mount1
[start] nfsexport:Dummy Export
[start] nfsclient:Admin group
[start] nfsclient:User group
[start] nfsclient:red
[start] script:initscript
[start] clusterfs:argle
Start of test1 complete
Starting test2...
[start] service:test2
[start] script:initscript
[start] clusterfs:argle
[start] ip:192.168.1.3
[start] fs:mount2
[start] nfsexport:Dummy Export
[start] nfsclient:Admin group
[start] nfsclient:User group
[start] nfsclient:red
[start] script:script2
[start] ip:192.168.1.4
[start] script:script3
Start of test2 complete
